src/queue.c

```diff
@@ -205,8 +347,43 @@ queue_get(Queue q, Message msg)
         return ret;
     }
 
-    firstrec = q->first;
-    q->first = q->first->next;
+    abstimeout.tv_sec  = now.tv_sec + q->timeout.tv_sec;
+    abstimeout.tv_nsec = (now.tv_usec*1000) + q->timeout.tv_nsec;
+    if (abstimeout.tv_nsec >= 1000000000)
```

Comment:
How big does  that go before it overflows?

Reply:
timespec.tv_nsec is a long. I did the maths, if we overshoot by ~2924 years we get an overflow.